src/Cache/Adapter/Redis.php

```diff
@@ -47,34 +50,55 @@ public function load(string $key, int $ttl): mixed
 
     /**
      * @param  string  $key
-     * @param  string|array<int|string, mixed>  $data
+     * @param  array<int|string, mixed>|string  $data
+     * @param  string  $hash optional
      * @return bool|string|array<int|string, mixed>
      */
-    public function save(string $key, $data): bool|string|array
+    public function save(string $key, array|string $data, string $hash = ''): bool|string|array
     {
-        if (empty($key) || empty($data)) {
+        if (empty($key) || empty($data) || empty($hash)) {
             return false;
         }
 
-        $cache = [
+        $value = json_encode([
             'time' => \time(),
             'data' => $data,
-        ];
+        ]);
+
+        if (! $value) {
```

Comment:
You can remove this check if you add to json_encode this flag 
\json_encode([], flags: JSON_THROW_ON_ERROR);
and put it in the try catch block below

Reply:
Done